### ISSUE TITLE
QCircuit.CTRL() with multiple targets throws "qpp::applyCTRL(): Matrix mismatch subsystems! [A/target]"

### ISSUE THREAD

**Issue by antoine-bussy** [softwareQinc/qpp#130]
Hi,

Executing the following circuit
```c++
auto const circuit = qpp::QCircuit{ 3u } .CTRL(qpp::gt.X, { 0u }, { 1u, 2u });
```
throws `"qpp::applyCTRL(): Matrix mismatch subsystems! [A/target]"`

I've implemented a unit test: https://github.com/antoine-bussy/qpp/commit/fd15ca8936bb15f4c8a90451b2aa0972b7349156
I suspect that is treated as a `CTRL_joint` in `QEngine::execute`.

**Comment by antoine-bussy** [comment]
Replacing `U` with `kronpow(U, target.size())` fixes the issue. See https://github.com/antoine-bussy/qpp/commit/6da00f502e4a7be9bfd91452597ad2e67b50efa3
But I guess it's not a very good fix performance-wise.

**Comment by vsoftco** [comment]
Thanks. The function is intended to apply a control gate, with control {0}, and target {1,2} in this example. So the target contains 2 qubits, hence the matrix you're applying need to be a 2-qubit gate (4 x 4 matrix). `gt.X` is a 1 qubit gate, hence a 2 x 2 matrix. This is not the same as a a control-X-X-...X

**Comment by antoine-bussy** [comment]
The documentation of `QCircuit::CTRL` states that `U` is a single qubit gate, unlike `QCircuit::CTRL_joint`.
Have I misunderstood the issue https://github.com/softwareQinc/qpp/issues/97 ?

**Comment by vsoftco** [comment]
We will correct the documentation. U can be a multiple-qubit gate. CTRL-joint has multiple controls, but same as CTRL for target (i.e., a multi-qubit gate)

**Comment by vsoftco** [comment]
Fixed by https://github.com/softwareQinc/qpp/releases/tag/v4.0-rc1

**Comment by vsoftco** [comment]
Fixed by #135
